pulpcore/pulpcore/tasking/services/worker_watcher.py

```diff
@@ -76,17 +77,20 @@ def check_celery_processes():
     now = timezone.now()
     oldest_heartbeat_time = now - timedelta(seconds=TASKING_CONSTANTS.PULP_PROCESS_TIMEOUT_INTERVAL)
 
-    for worker in Worker.objects.filter(last_heartbeat__lt=oldest_heartbeat_time, online=True):
+    for worker in Worker.objects.filter(last_heartbeat__lt=oldest_heartbeat_time,
+                                        gracefully_stopped=False):
         msg = _("Worker '%s' has gone missing, removing from list of workers") % worker.name
         _logger.error(msg)
 
         mark_worker_offline(worker.name)
 
     worker_count = Worker.objects.exclude(
-        name__startswith=TASKING_CONSTANTS.RESOURCE_MANAGER_WORKER_NAME).filter(online=True).count()
+        name__startswith=TASKING_CONSTANTS.RESOURCE_MANAGER_WORKER_NAME)\
+        .filter(gracefully_stopped=False).count()
 
     resource_manager_count = Worker.objects.filter(
-        name__startswith=TASKING_CONSTANTS.RESOURCE_MANAGER_WORKER_NAME).filter(online=True).count()
+        name__startswith=TASKING_CONSTANTS.RESOURCE_MANAGER_WORKER_NAME)\
```

Comment:
Similar for this, can this be refactored to use `Worker.objects.online_workers()`. I think the current approach would include non-gracefully stopped workers incorrectly without a change.

Reply:
Again nope, for the same reason.  It's only counting resource managers.